metrics/spearmanr/README.md

```diff
@@ -0,0 +1,117 @@
+# Metric Card for Spearman Correlation Coefficient Metric (spearmanr)
+
+
+## Metric Description
+The Spearman rank-order correlation coefficient is a measure of the
+relationship between two datasets. Like other correlation coefficients,
+this one varies between -1 and +1 with 0 implying no correlation.
+Positive correlations imply that as data in dataset x increases, so 
+does data in dataset y. Negative correlations imply that as x increases,
+y decreases. Correlations of -1 or +1 imply an exact monotonic relationship.
+
+Unlike the Pearson correlation, the Spearman correlation does not 
+assume that both datasets are normally distributed. 
+
+The p-value roughly indicates the probability of an uncorrelated system
+producing datasets that have a Spearman correlation at least as extreme
+as the one computed from these datasets. The p-values are not entirely
+reliable but are probably reasonable for datasets larger than 500 or so.
+
+
+## How to Use
+At minimum, this metric only requires a `list` of predictions and a `list` of references:
+
+```python
+>>> spearmanr_metric = datasets.load_metric("spearmanr")
+>>> results = spearmanr_metric.compute(references=[1, 2, 3, 4, 5], predictions=[10, 9, 2.5, 6, 4])
+>>> print(results)
+{'spearmanr': -0.7}
+```
+
+### Inputs
+- **`predictions`** (`list` of `float`): Predicted labels, as returned by a model.
+- **`references`** (`list` of `float`): Ground truth labels.
+- **`return_pvalue`** (`bool`): If `True`, returns the p-value. If `False`, returns
+                    only the spearmanr score. Defaults to `False`.
+
+### Output Values
+-  **`spearmanr`** (`float`): Spearman correlation coefficient.
+- **`p-value`** (`float`): p-value. **Note**: is only returned
+                        if `return_pvalue=True` is input.
+
+If `return_pvalue=False`, the output is a `dict` with one value, as below:
+```python
+{'spearmanr': -0.7}
+```
+
+Otherwise, if `return_pvalue=True`, the output is a `dict` containing a `SpearmanrResult` object:
+```python
+{'spearmanr': SpearmanrResult(correlation=-0.7, pvalue=0.1881204043741873)}
+```
+
+Spearman rank-order correlations can take on any value from `-1` to `1`, inclusive.
+
+The p-values can take on any value from `0` to `1`, inclusive.
+
+#### Values from Popular Papers
+
+
+### Examples
+A basic example:
+```python
+>>> spearmanr_metric = datasets.load_metric("spearmanr")
+>>> results = spearmanr_metric.compute(references=[1, 2, 3, 4, 5], predictions=[10, 9, 2.5, 6, 4])
+>>> print(results)
+{'spearmanr': -0.7}
+```
+
+The same example, but that also returns the pvalue:
+```python
+>>> spearmanr_metric = datasets.load_metric("spearmanr")
+>>> results = spearmanr_metric.compute(references=[1, 2, 3, 4, 5], predictions=[10, 9, 2.5, 6, 4], return_pvalue=True)
+>>> print(results)
+{'spearmanr': SpearmanrResult(correlation=-0.7, pvalue=0.1881204043741873)}
+>>> print(results['spearmanr'].correlation)
+-0.7
+>>> print(round(results['spearmanr'].pvalue, 2))
+0.19
```

Comment:
(nit) I would have expected it to return a "pvalue" field with the actual value inside it. Is there a reason to keep everything ?

Reply:
I think keeping both values is useful, depending on what people use this for!